Integrations/DomainTools_Iris/iris.py

```diff
@@ -0,0 +1,393 @@
+
+
+def domain_profile():
+    res = http_request('GET', '/v1/iris-investigate/', {'domain': demisto.args().get('domain')})
+    if res.get('results_count'):
+        human_readable, outputs = format_domain_profile(res)
+        return_outputs(human_readable, outputs, res)
+    else:
+        demisto.results('No results found.')
+
+
+def domain_analytics():
+    res = http_request('GET', '/v1/iris-investigate/', {'domain': demisto.args().get('domain')})
+    if res.get('results_count'):
+        human_readable, outputs = format_domain_analytics(res)
+        return_outputs(human_readable, outputs, res)
+    else:
+        demisto.results('No results found.')
+
+
+def threat_profile():
+    res = http_request('GET', '/v1/iris-investigate/', {'domain': demisto.args().get('domain')})
+    if res.get('results_count'):
+        human_readable, outputs = format_threat_profile(res)
+        return_outputs(human_readable, outputs, res)
+    else:
+        demisto.results('No results found.')
+
+
+def domain_pivot():
+    res = None  # type: ignore
+    search_type = ""
+    search_value = ""
+    if demisto.args().get('ip'):
+        res = http_request('GET', '/v1/iris-investigate/', {'ip': demisto.args().get('ip')})
+        search_type, search_value = 'IP', demisto.args().get('ip')
+    elif demisto.args().get('email'):
+        res = http_request('GET', '/v1/iris-investigate/', {'email': demisto.args().get('email')})
+        search_type, search_value = 'E-Mail', demisto.args().get('email')
+    elif demisto.args().get('nameserver_ip'):
+        res = http_request('GET', '/v1/iris-investigate/', {'nameserver_ip': demisto.args().get('nameserver_ip')})
+        search_type, search_value = 'Name Server IP', demisto.args().get('nameserver_ip')
+    elif demisto.args().get('ssl_hash'):
+        res = http_request('GET', '/v1/iris-investigate/', {'ssl_hash': demisto.args().get('ssl_hash')})
+        search_type, search_value = 'SSL Hash', demisto.args().get('ssl_hash')
+    elif demisto.args().get('nameserver_host'):
+        res = http_request('GET', '/v1/iris-investigate/', {'nameserver_host': demisto.args().get('nameserver_host')})
+        search_type, search_value = 'Name Server Host', demisto.args().get('nameserver_host')
+    elif demisto.args().get('mailserver_host'):
+        res = http_request('GET', '/v1/iris-investigate/', {'mailserver_host': demisto.args().get('mailserver_host')})
+        search_type, search_value = 'Mail Server Host', demisto.args().get('mailserver_host')
+
+    if res.get('results_count'):
+        human_readable, outputs = format_domain_pivot(res, search_type, search_value)
+        return_outputs(human_readable, outputs, res)
+    else:
+        demisto.results('No results found.')
+
+
+def test_module():
+    res = http_request('GET', '/v1/iris-investigate/', {'domain': 'demisto.com'})
+    if res.get('results_count'):
+        demisto.results('ok')
+    else:
+        demisto.results('No results found.')
+
+
+try:
```

Comment:
Wrap this in a `main` function

Reply:
Do you have an example of this? I don't see any other integration doing this.

Reply:
Yes, many.
Look at [Proofpoint](https://github.com/demisto/content/tree/master/Integrations/ProofpointTAP_v2) for one

Reply:
When I wrap it in main the code no longer works in demisto. The test module can't run and all the other commands leave dbot hanging. Is there anything special you have to do to make it work with wrapping in main?

Reply:
Did you add these 2 lines to the code:
```
if __name__ in ('__main__', 'builtin', 'builtins'):
    main()
```

Reply:
Yes.

Reply:
Can you push the code so I can see it?

Reply:
Code pushed.

Reply:
im able to get results with the following code:
```
def main():
    """
    Main Demisto function.
    """
    try:
        if demisto.command() == 'test-module':
            test_module()
        elif demisto.command() == 'domain':
            domain_profile_command()
        elif demisto.command() == 'domaintoolsiris-analytics':
            domain_analytics_command()
        elif demisto.command() == 'domaintoolsiris-threat-profile':
            threat_profile_command()
        elif demisto.command() == 'domaintoolsiris-pivot':
            domain_pivot_command()
    except Exception as e:
        return_error('Unable to perform command : {}, Reason: {}'.format(demisto.command(), str(e)))

if __name__ in ['__main__', 'builtin', 'builtins']:
    main()
```
![image](https://user-images.githubusercontent.com/31018228/66660951-c3514c00-ec4e-11e9-9cf3-2e8b4c6d8ea8.png)

Reply:
Ya I have that and the test module fails and the integration commands never return. Would version of Demisto or anything like that affect it? I'm on version 4.5.0 Build 40783

Reply:
Demisto version shouldn't matter.
What does the test module fail on?

Reply:
I get this message when I click the test button in the integration settings. ![image](https://user-images.githubusercontent.com/737690/66752186-b3e82380-ee45-11e9-9016-b010c83da10d.png)

It works fine wrapping in main.